### PR TITLE
[Move] Implement Flame Burst

### PR DESCRIPTION
**Pull request by torranx**
## What are the changes?
- implement flame burst

## Why am I doing these changes?
- unimplemented move

## What did change?
- new class `FlameBurstAttr`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -1429,6 +1429,34 @@ export class PartyStatusCureAttr extends MoveEffectAttr {
...
+   * @returns A boolean indicating whether the effect was successfully applied.
+   */
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean | Promise<boolean> {
+    const targetAlly = target.getAlly();
+
+    if (!targetAlly) {
+      return false;
+    }
+
+    targetAlly.damageAndUpdate(Math.floor(1/16 * targetAlly.getMaxHp()), HitResult.OTHER);
```
Comment (schmidtc1) [inline src/data/move.ts:1451]: Very niche case, but how does this work against Innards Out.
E.g. Enemy 2 has Innards Out, Flame Burst on Enemy 1 -> Splash damage on Enemy 2 to faint it. Should Innards Out trigger on the Flame Burst user?

Comment (torranx) [reply src/data/move.ts:1451]: I don't think Innards Out would take effect from effect damage- bulba also says:
> ...knocked out by an opponent's attack.
The damage is not directly from an opponent's attack.

File: src/data/move.ts [hunk 2]
```diff
@@ -1429,6 +1429,40 @@ export class PartyStatusCureAttr extends MoveEffectAttr {
...
+    if (cancelled.value || !targetAlly) {
+      return false;
+    }
+
+    targetAlly.damageAndUpdate(Math.max(1, Math.floor(1/16 * targetAlly.getMaxHp())), HitResult.OTHER);
+    return true;
+  }
+
+  getTargetBenefitScore(user: Pokemon, target: Pokemon, move: Move): integer {
+    return target.getAlly() ? -5 : 0;
```
Comment (Tempo-anon) [inline src/data/move.ts:1462]: It's -5 if the target has an ally but 0 if they don't?

Comment (torranx) [reply src/data/move.ts:1462]: ikr this is quite confusing: (`getTargetBenefitScore`) negative target scores = more incentive to use it against the enemy

discussed here: https://discord.com/channels/1125469663833370665/1176874654015684739/1265796251044155587

Comment (Tempo-anon) [reply src/data/move.ts:1462]: That should probably be documented somewhere because that's sort of confusing

**Review by Tempo-anon — APPROVED** [review]

**Pull request closed without merging**